lint/util.py

```diff
@@ -279,7 +279,14 @@ def communicate(cmd, code=None, output_stream=STREAM_STDOUT, env=None, cwd=None,
         if output_stream == STREAM_STDERR:
             return _post_process_fh(out[1])
         if output_stream == STREAM_BOTH:
-            return ''.join(_post_process_fh(fh) for fh in out)
+            if _linter and hasattr(_linter, 'on_stderr'):
+                stdout, stderr = [_post_process_fh(fh) for fh in out]
+                if stderr:
+                    _linter.on_stderr(stderr)
```

Comment:
Calling this from within `util.communicate` seems bad practice. I'd rather return a `stderr, stdout` tuple always (with either as `None` if not requested) and let some code inside `Linter` take care of handling the different streams, i.e. passing them around. 

Downside is that it changes the function signature and some plugins may use it. I know mypy does.

Ideally, we would add a second parameter to `find_errors` to hand the streams separately there, but I suppose that would be too breaking?

Reply:
Calling this from `util` is not an issue. I already refactored it from both sides, so that I have this ugly `_linter` arg. I can move this to `Linter` flip the arguments and call `_linter` `self`. 

Now, initially you think you want full control and everything ready in `find_errors` or something similar. But then, should every plugin implement `find_errors` to get proper failure messaging? Probably not, so the question was if we are good to go with a simple 'callback' fn. 

We can make this opt-in via `hasattr` and if we get more knowledge about the different linters, we probably can make a trivial `on_stderr` the default.

The trivial impl is 

```
def on_stderr(self, output):
	logger.error(output)
	self.notify_failure()
```

Reply:
Ah well, considering that usually linter output happens on stdout and stderr is for usage instructions, I guess special-casing the stderr stream like this is okay.

You know more about how linter plugins interact with the core, so I'll retract my concerns. These are just the kind of things that cause me to think "maybe I structured this suboptimally", but I'm not keen on bothering everyone with that.